lib/hammer_cli_csv/content_hosts.rb

```diff
@@ -348,12 +351,12 @@ def update_subscriptions(host, line, remove_existing)
         })['results']
         if remove_existing && existing_subscriptions.length != 0
           existing_subscriptions.map! do |existing_subscription|
-            {:id => existing_subscription['id'], :quantity => existing_subscription['quantity_consumed']}
+            subs = [{:id => existing_subscription['id'], :quantity => existing_subscription['quantity_consumed']}]
+            @api.resource(:host_subscriptions).call(:remove_subscriptions, {
+              'host_id' => host['id'],
+              'subscriptions' => subs
+            })
           end
-          @api.resource(:host_subscriptions).call(:remove_subscriptions, {
-            'host_id' => host['id'],
-            'subscriptions' => existing_subscriptions
-          })
```

Comment:
There was a race condition in candlepin, solved but safest to support older versions of foreman.
https://github.com/Katello/katello/pull/6722